sources/views/contactsData/contactForm.js

```diff
@@ -0,0 +1,269 @@
+							autosend: false,
+							multiple: false,
+							width: 150,
+							on: {
+								onBeforeFileAdd: (item) => {
+									const fileReader = new FileReader();
+									fileReader.readAsDataURL(item.file);
+									fileReader.onload = () => this.$$("contactPhoto").setValues({Photo: fileReader.result});
+								},
+								onFileUploadError: () => {
+									webix.message({type: "error", text: "Error during uploading photo!"});
+								}
+							}
+						},
+						{
+							view: "button",
+							value: "Delete photo",
+							width: 150,
+							click: () => this.deletePhoto()
+						}
+					]
+				}
+			]
+		};
+
+		const contactFormButtons = {
+			cols: [
+				{},
+				{
+					view: "button",
+					value: "Cancel",
+					width: 150,
+					click: () => this.cancelForm()
+				},
+				{
+					view: "button",
+					localId: "addSaveButton",
+					width: 150,
+					click: () => this.addSaveContact()
+				}
+			]
+
+		};
+		return {
+			view: "form",
+			localId: "contactForm",
+			margin: 20,
+			elements: [
+				headContactForm,
+				{
+					cols: [
+						contactLeftForm,
+						{
+							rows: [
+								contactRightForm,
+								contactPhotoForm
+							]
+						}
+					]
+				},
+				{},
+				contactFormButtons
+			],
+			rules: {
+				FirstName: this.webix.rules.isNotEmpty,
+				LastName: this.webix.rules.isNotEmpty,
+				StartDate: this.webix.rules.isNotEmpty,
+				Company: this.webix.rules.isNotEmpty,
+				Phone: this.webix.rules.isNotEmpty,
+				Birthday: this.webix.rules.isNotEmpty
+			}
+		};
+	}
+
+	init() {
+		this.contactForm = this.$$("contactForm");
+		this.headContactForm = this.$$("headContactForm");
+		this.addSaveButton = this.$$("addSaveButton");
+		this.contactPhoto = this.$$("contactPhoto");
+
+		const contactId = this.getParam("id", true);
+
+		const selectedAction = contactId ? "Edit" : "Add";
+		this.headContactForm.setValues({selectedAction});
+
+		const addOrSave = contactId ? "Save" : "Add";
+		this.addSaveButton.setValue(addOrSave);
+
+		if (contactId && contacts.exists(contactId)) {
+			let contactValues;
+			contacts.waitData
+				.then(() => {
+					contactValues = contacts.getItem(contactId);
+					this.contactForm.setValues(contactValues);
+					this.contactPhoto.setValues({Photo: contactValues.Photo});
+				});
+		}
+	}
+
+	cancelForm(id) {
+		this.contactForm.clear();
+		this.contactForm.clearValidation();
+		this.show(`/top/contacts?id=${id}/contactsData.contactsTemplate`);
+	}
+
+	addSaveContact() {
+		contacts.waitSave(() => {
+			if (this.contactForm.validate()) {
+				const values = this.contactForm.getValues();
+				if (values && values.id) {
+					contacts.updateItem(values.id, values);
+				}
+				else {
+					contacts.add(values, 0);
+				}
+			}
+		})
+			.then((res) => {
+				if (res.length === 0) {
+					return;
+				}
+				this.app.callEvent("onContactSelect", [res]);
+				this.cancelForm(res.id);
+			});
+	}
+
+	deletePhoto() {
+		this.contactPhoto.setValues({Photo: avatarDefault});
```

Comment:
Do not set `avatarDefault` here, as in such case it may be saved to the database. Just set an empty object (better option) or `{ Photo:"" }`.

Reply:
fixed